src/test/regress/expected/multi_with_join.out

```diff
@@ -396,4 +404,193 @@ LIMIT
        2 | Wed Nov 22 18:19:49.944985 2017 |       1
        2 | Wed Nov 22 18:19:49.944985 2017 |       1
 (10 rows)
+-- cte JOIN reference_table should be router plannable
+EXPLAIN (COSTS false, VERBOSE true)
+WITH cte AS (
+  SELECT * FROM users_table
+)
+SELECT * FROM cte join reference_table ON cte.user_id + 1 = reference_table.user_id;
+                                                                                                             QUERY PLAN                                                                                                             
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ Custom Scan (Citus Router)
+   Output: remote_scan.user_id, remote_scan."time", remote_scan.value_1, remote_scan.value_2, remote_scan.value_3, remote_scan.value_4, remote_scan.user_id_1
+   ->  Distributed Subplan
```

Comment:
The explain output is going to be pretty noisy and will change randomly. We could maybe use some SQL feature that is only available in the router planner to verify that it's a router plan, alternatively we could turn on debug output.